feat(apple): Add tracePropagationTarget to automatic instrumentation (#7160)
getsentry/sentry-docs: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx

[NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
[NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection

### Trace Propagation Targets

> Sentry addes an extra header with the trace id in the outgoing HTTP requests in order to continue the transaction in the backend.

Reviewer: ```suggestion
Sentry adds an extra header with the trace id in the outgoing HTTP requests to continue the transaction in the backend.
```

> It's possible to filter out which requests Sentry should add the extra header by setting the `tracePropagationTarget` option, this way, only the app backend will receive the trace id.

Reviewer: ```suggestion
You can set the `tracePropagationTarget` option to filter which requests Sentry adds the extra header to.
For example, to ensure that only the app backend will receive the trace id.
```

```swift {tabTitle:Swift}
import Sentry

SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.tracePropagationTargets = ["MyAppDomain.com"]
}
```

```objc {tabTitle:Objective-C}
@import Sentry;

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracePropagationTargets = @[@"MyAppDomain.com"]
}];
```

The option may contain a list of NSString or NSRegularExpression against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.

> If tracePropagationTargets is not provided, trace data is attached to every outgoing request from the instrumented client.

Reviewer: ```suggestion
If `tracePropagationTargets` is not provided, trace data is attached to every outgoing request from the instrumented client.
```

[NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
[NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection

### Trace Propagation Targets

Sentry adds an extra header with the trace id in the outgoing HTTP requests to continue the transaction in the backend.

You can set the `tracePropagationTarget` option to filter which requests Sentry adds the extra header to.

> For example, to ensure that only the app backend will receive the trace id.

Reviewer 2: `l`:
```suggestion
For example, to ensure that only your app backend will receive the trace id.
```

```swift {tabTitle:Swift}
import Sentry

SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.tracePropagationTargets = ["MyAppDomain.com"]
}
```

```objc {tabTitle:Objective-C}
@import Sentry;

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracePropagationTargets = @[@"MyAppDomain.com"]
}];
```

> The option may contain a list of NSString or NSRegularExpression against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.

Reviewer 2: `l`:
```suggestion
The option may contain a list of `NSString` or `NSRegularExpression` against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.
```
